src/document.h

```diff
@@ -708,6 +708,22 @@ struct Document {
         return ExportFile(fn, k, true);
     }
 
+    wxBitmap GetBitmap() {
+        maxx = layoutxs;
+        maxy = layoutys;
+        originx = originy = 0;
+        wxBitmap bm(maxx, maxy, 24);
+        wxMemoryDC mdc(bm);
+        DrawRectangle(mdc, Background(), 0, 0, maxx, maxy);
+        Render(mdc);
+        return bm;
+    }
+
+    wxBitmap GetSubBitmap(Selection &s) {
+        wxRect r = s.g->GetRect(this, s, true);
+        return GetBitmap().GetSubBitmap(r);

```

Comment:
Ok, this is not great, since you're rendering the whole view then only copying part of it.. would be better to modify GetBitmap to work correctly both with the view root and a selection.. but if its more complicated, I guess this only happens when you explicitly use copy as bitmap, I guess its ok for now.

Reply:
Thanks for your reply! I totally see your point, maybe I will find a way to have a better way. Render() seems to have something in document.h:533.